### PR TITLE
Bugfix limit participant events to account

### PR DESCRIPTION
**Pull request by jamhed**

### CODE REVIEW

**Comment by fenollp** [comment]
elvis says there're trailing whitespace in bh_conference

**Comment by jamhed** [comment]
WIP: other blackhole modules needs to be converted too.

File: applications/blackhole/src/modules/bh_conference.erl [hunk 1]
```diff
@@ -25,32 +25,57 @@ handle_event(Context, EventJObj) ->
...
+        [<<"*">>, <<"*">>] ->
+            event_unbind_all(Context, AccountDb);
+        [<<"*">>, _CallId] ->
+            lager:debug("invalid conference event bind: ~s", [Binding]);
         [ConfId, CallId] ->
-            blackhole_listener:remove_binding('conference', event_binding_options(ConfId, CallId));
-        _Else -> lager:debug("invalid conference event unbind: ~s", [Binding])
+            event_unbind(Context, ConfId, CallId);
+        _Else ->
+            lager:debug("invalid conference event bind: ~s", [Binding])
```
Comment (k-anderson) [inline applications/blackhole/src/modules/bh_conference.erl]: @jamhed Are there errors sent back for these invalid bind attempts?

File: applications/blackhole/src/modules/bh_conference.erl [hunk 2]
```diff
@@ -68,3 +93,63 @@ event_binding_options(ConfId, CallId) ->
     ,{'restrict_to', ['event']}
     ,'federate'
     ].
+
+%% commands
+
+command_bind(Context, ConfId) ->
```
Comment (k-anderson) [inline applications/blackhole/src/modules/bh_conference.erl]: @jamhed specs ;)

File: applications/blackhole/src/modules/bh_conference.erl (hunk 1 above)
Comment (jamhed) [inline applications/blackhole/src/modules/bh_conference.erl]: @k-anderson, not at the moment.

File: applications/blackhole/src/modules/bh_conference.erl (hunk 2 above)
Comment (jamhed) [inline applications/blackhole/src/modules/bh_conference.erl]: @k-anderson, sure.

**Comment by k-anderson** [comment]
@jamhed Bump ;)

File: core/kazoo/src/api/kapi_conference.erl [hunk 3]
```diff
@@ -1294,13 +1294,13 @@ publish_participant_volume_out(ConferenceId, Req, ContentType) ->
...
-    publish_participant_event(ConferenceId, CallId, JObj, ?DEFAULT_CONTENT_TYPE).
-publish_participant_event(ConferenceId, CallId, Event, ContentType) ->
+-spec publish_participant_event(ne_binary(), ne_binary(), ne_binary(), api_terms()) -> 'ok'.
+-spec publish_participant_event(ne_binary(), ne_binary(), ne_binary(), api_terms(), ne_binary()) -> 'ok'.
+publish_participant_event(AccountId, ConferenceId, CallId, JObj) ->
+    publish_participant_event(AccountId, ConferenceId, CallId, JObj, ?DEFAULT_CONTENT_TYPE).
+publish_participant_event(AccountId, ConferenceId, CallId, Event, ContentType) ->
     {'ok', Payload} = kz_api:prepare_api_payload(Event, ?PARTICIPANT_EVENT_VALUES, fun ?MODULE:participant_event/1),
-    amqp_util:conference_publish(Payload, 'event', ConferenceId, amqp_util:encode(CallId), [], ContentType).
+    amqp_util:conference_publish(Payload, 'event', AccountId, ConferenceId, amqp_util:encode(CallId), [], ContentType).
```
Comment (lazedo) [inline core/kazoo/src/api/kapi_conference.erl]: @jamhed can't we retrieve the account-id from the payload ?

File: core/kazoo_amqp/src/amqp_util.erl [hunk 4]
```diff
@@ -392,9 +392,9 @@ conference_publish(Payload, 'event', ConfId, Options, ContentType) ->
 conference_publish(Payload, 'command', ConfId, Options, ContentType) ->
     basic_publish(?EXCHANGE_CONFERENCE, <<?KEY_CONFERENCE_COMMAND/binary, ConfId/binary>>, Payload, ContentType, Options).
 
-conference_publish(Payload, 'event', ConfId, CallId, Options, ContentType) ->
+conference_publish(Payload, 'event', AccountId, ConfId, CallId, Options, ContentType) ->
```
Comment (lazedo) [inline core/kazoo_amqp/src/amqp_util.erl]: @jamhed same here

File: applications/ecallmgr/src/ecallmgr_fs_conference.erl [hunk 5]
```diff
@@ -248,15 +248,21 @@ code_change(_OldVsn, State, _Extra) ->
 %%%===================================================================
 %%% Internal functions
 %%%===================================================================
+-spec finalize_processing(ne_binary(), ne_binary()) -> 'ok'.
+finalize_processing(<<"del-member">>, CallId) ->
+    ecallmgr_fs_conferences:participant_destroy(CallId),
+    'ok';
+finalize_processing(_, _) -> 'skip'.
```
Comment (lazedo) [inline applications/ecallmgr/src/ecallmgr_fs_conference.erl]: shouldn't this be 'ok' to conform to spec ?

File: applications/ecallmgr/src/ecallmgr_fs_conference.erl [hunk 6]
```diff
@@ -798,13 +798,13 @@ relay_event(UUID, Node, Props) ->
 
 -spec publish_participant_event(boolean(), kz_proplist(), ne_binary(), kz_proplist()) -> ok | skip.
 publish_participant_event(true=_Publish, Event, CallId, Props) ->
-    Ev = [{<<"Event-Category">>, <<"conference">>}
-         ,{<<"Event-Name">>, <<"participant_event">>}
-          | Event],
     #participant{call_info=CCV} = ecallmgr_fs_conferences:participant_get(CallId),
+    Ev = props:set_value(<<"Custom-Channel-Vars">>, CCV,
+                         [{<<"Event-Category">>, <<"conference">>}
+                         ,{<<"Event-Name">>, <<"participant_event">>} | Event]),
```
Comment (lazedo) [inline applications/ecallmgr/src/ecallmgr_fs_conference.erl]: @jamhed i meant this
```
Ev = [{<<"Event-Category">>, <<"conference">>}
     ,{<<"Event-Name">>, <<"participant_event">>}
     ,{<<"Custom-Channel-Vars">>, CCV}
     | Event
     ]),
```
sorry if i didn't explained myself better

**Pull request merged**